public/scripts/week-calendar.js

```diff
@@ -88,6 +88,39 @@ function populateRemindersModal(userId, reminderId) {
   });
 }
 
+function populateTasksModal(userId, taskId) {
+  $.ajax({
+    method: "GET",
+    url: `/task/${taskId}`,
+    success: function (data) {
+      dataGlobal = data;
+      userIdGlobal = userId;
+
+      let event_modal = document.getElementById("modal-task-display");
+
+      // event_modal.querySelector("#modal-task-label.modal-title").innerText =
+      //   data.title;
+      event_modal.querySelector("input#task_title").value = data.title;
+      event_modal.querySelector("input#task_textBody").value = data.textBody;
+      event_modal.querySelector("input#task_tag").value = data.tag;
+      event_modal.querySelector("select#task_priority").value = data.priority;
+      // issue with date time coming as a date string
+      // it needs an iso string
+      event_modal.querySelector("input#task_dateAddedTo").value =
+        data.dateAddedTo;
+
+      event_modal.querySelector("input#task_dateDueOn").value = data.dateDueOn;
```

Comment:
lets keep just dateAddedto for task since they are always due for the same date. we can discuss about this